File: user-service/src/main/java/rs/raf/user_service/client/BankClient.java
package rs.raf.user_service.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;
import rs.raf.user_service.domain.dto.CardRequestDto;

@FeignClient(name = "bank-service", url = "${spring.cloud.openfeign.client.config.bank-service.url}", fallbackFactory = BankClientFallbackFactory.class)
public interface BankClient {

    @PostMapping("/api/payment/confirm-payment/{id}")
    void confirmPayment(@PathVariable("id") Long id);

    @PostMapping("/api/payment/confirm-transfer/{id}")
    void confirmTransfer(@PathVariable("id") Long id);

    @PutMapping("/api/account/{id}/change-limit")
    void changeAccountLimit(@PathVariable("id") Long id);

    @PutMapping("/api/account/1/cards/approve/{id}")
    void approveCardRequest(@PathVariable("id") Long id);

    @PostMapping("/api/payment/reject-payment/{id}")
    void rejectConfirmPayment(@PathVariable("id") Long id);

    @PostMapping("/api/payment/reject-transfer/{id}")
    void rejectConfirmTransfer(@PathVariable("id") Long id);//

    @PutMapping("/api/account/{id}/change-limit/reject")
    void rejectChangeAccountLimit(@PathVariable("id") Long id);//

    @PutMapping("/api/account/1/cards/reject/{id}")
    void rejectApproveCardRequest(@PathVariable("id") Long id);//

}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/external/BankClient.kt
package pack.userservicekotlin.external

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping

@FeignClient(
    name = "bank-service",
    url = "\${spring.cloud.openfeign.client.config.bank-service.url}",
    fallbackFactory = BankClientFallbackFactory::class,
)
interface BankClient {
    @PostMapping("/api/payment/confirm-payment/{id}")
    fun confirmPayment(
        @PathVariable("id") id: Long?,
    )

    @PostMapping("/api/payment/confirm-transfer/{id}")
    fun confirmTransfer(
        @PathVariable("id") id: Long?,
    )

    @PutMapping("/api/account/{id}/change-limit")
    fun changeAccountLimit(
        @PathVariable("id") id: Long?,
    )

    @PutMapping("/api/account/1/cards/approve/{id}")
    fun approveCardRequest(
        @PathVariable("id") id: Long?,
    )

    @PostMapping("/api/payment/reject-payment/{id}")
    fun rejectConfirmPayment(
        @PathVariable("id") id: Long?,
    )

    @PostMapping("/api/payment/reject-transfer/{id}")
    fun rejectConfirmTransfer(
        @PathVariable("id") id: Long?,
    ) //

    @PutMapping("/api/account/{id}/change-limit/reject")
    fun rejectChangeAccountLimit(
        @PathVariable("id") id: Long?,
    ) //

    @PutMapping("/api/account/1/cards/reject/{id}")
    fun rejectApproveCardRequest(
        @PathVariable("id") id: Long?,
    ) //
}
